src/content.ts

```diff
@@ -84,9 +72,24 @@ function listen(elem) {
     )
 }
 listen(window)
-document.addEventListener("readystatechange", _ =>
-    getAllDocumentFrames().forEach(f => listen(f)),
-)
+
+async function iframeMutationCallback(mutations: MutationRecord[], observer: MutationObserver) {
+    const leavegithubalone = await config.getAsync("leavegithubalone")
+
+    for (const mutation of mutations) {
+        if (mutation.type === "childList")
+            for (const node of mutation.addedNodes)
+                if ((node instanceof HTMLIFrameElement) || (node instanceof HTMLFrameElement)) {
+                    listen(node.contentDocument);
+                    observer.observe(node.contentDocument, {subtree: true, childList: true})
+                    if (leavegithubalone !== "true")
+                        node.contentDocument.body.addEventListener("keydown", protectSlash)
+                }
+    }
+}
+
+const iframeMutationObserver = new MutationObserver(iframeMutationCallback)
+iframeMutationObserver.observe(document, {subtree: true, childList: true})
```

Comment:
I'm not sure what guarantees Firefox makes about the order of execution. This observer is set up as soon as navigation happens, which from my brief experiments is enough to consistently catch all changes done from loading the actual page. It might not be bulletproof though, especially if we start using top level async-await.